airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -398,4 +329,141 @@ private void continueAsNew(final ConnectionUpdaterInput connectionUpdaterInput)
     }, input);
   }
 
+
+  /**
+   * Return the duration to wait. This is calculated by the configFetchActivity and return the
+   * duration to wait until the next run
+   */
+  private Duration getTimeToWait(UUID connectionId) {
+    // Scheduling
+    final ScheduleRetrieverInput scheduleRetrieverInput = new ScheduleRetrieverInput(connectionId);
+
+    final ScheduleRetrieverOutput scheduleRetrieverOutput = configFetchActivity.getTimeToWait(scheduleRetrieverInput);
+
+    return scheduleRetrieverOutput.getTimeToWait();
+  }
+
+  /**
+   * Check if a sync need to be run based on the workflow state.
+   */
+  private boolean needToProcessSync() {
+    return !workflowState.isUpdated() && !workflowState.isDeleted();
+  }
+
+  /**
+   * Creates a new job if it is not present in the input. If the jobId is specified in the input of
+   * the connectionManagerWorkflow, we will return it. Otherwise we will create a job and return its
+   * id.
+   */
+  private Long getOrCreateJobId(final ConnectionUpdaterInput connectionUpdaterInput) {
+    return Optional.ofNullable(connectionUpdaterInput.getJobId()).or(() -> {
```

Comment:
why are we using multiple optionals here? we don't seem to be making use of any present checks, which makes me think we can remove the optionals. lmk if I'm missing something